src/checkpoints.cpp

```diff
@@ -368,7 +368,7 @@ namespace Checkpoints
         // Relay checkpoint
         {
             LOCK(cs_vNodes);
-            BOOST_FOREACH(CNode* pnode, vNodes)
+            for (auto const& pnode : vNodes)
```

Comment:
I think it's fine to keep the type in cases where the type does not reduce readability. For example:
```
Readable type: CNode*
Unreadable type: std::function<void(const std::string&)>
```

There's no need to change all of these back but it could be good to keep in mind for the future.

Reply:
interesting yea i will keep that in mind. it appears travis failed on the FileManifest

Reply:
ill keep that in mind but i believe i tried with for auto wiht CNode* and it said too many arguments. vNodes is CNode* by the way so it automatically makes pnode CNode*

Reply:
vector<CNode*> vNodes;

Reply:
Yeah then it should work to do `for(const CNode* pnode : vNodes)` :)

Reply:
ah yea i did that because the auto pulls the CNode* part from CNode* vNodes to make CNode* pnode :)